File: docs/archive/layer2_examples/example_risky.py
"""
Example Python script — contains known data risk patterns for Layer 2 testing.
"""

import pandas as pd

# Load data — missing checks follow
df = pd.read_csv("evictions.csv")

# Immediately print head — no shape, na, dtype inspection
print(df.head())

# Second load — also missing checks
df2 = pd.read_csv("names.csv")

# Cast ZIP to int — leading zeros lost
df["zip"] = df["zip"].astype(int)

# Filter without dtype check
evictions = df[df["county"] == "Los Angeles"]

# Exclude total row but via fragile string match — Total row present
total = df[df["county"] != "Total"].groupby("county")["count"].sum()

# Mean with no range check and no median
avg_rent = df["rent"].mean()
print("Average rent:", avg_rent)

# groupby with no category inspection first
by_county = df.groupby("county")["evictions"].sum()

# Unexplained filter threshold
high_risk = df[df["rate"] > 0.15]

# Sentinel value filter
income_avg = df[df["income"] != -99]["income"].mean()

# Merge with no row count check
merged = df.merge(df2, on="county")

# Join on string key (county_name is not an ID)
merged2 = df.merge(df2, on="county_name")

# Left join with no unmatched check afterward
left = df.merge(df2, on="id", how="left")

# Hardcoded significance threshold
from scipy import stats
t, p = stats.ttest_ind(df["group_a"], df["group_b"])
if p < 0.05:
    print("Significant")

# Percentage with no denominator shown
pct = df["evicted"].sum() / len(df) * 100
print(f"Eviction rate: {pct:.1f}%")

# Aggregation with no null handling
total_evictions = df["evictions"].sum()

# pct_change with no base-year note
annual = df.groupby("year")["evictions"].sum().pct_change()

File: docs/archive/layer2_examples/example_risky.R
# Example R script — contains known data risk patterns for Layer 2 testing.

library(dplyr)
library(readr)

# Load without checking anything
df <- read_csv("evictions.csv")

# Just print head — no inspection
head(df)

# Second load — also no checks
df2 <- read.csv("names.csv")

# No dtype inspection
evictions <- df %>% filter(county == "Los Angeles")

# Cast ZIP to numeric — leading zeros lost
df$zip_num <- as.numeric(df$zip)

# Total row not excluded before groupby
by_county <- df %>%
  filter(county != "Total") %>%
  group_by(county) %>%
  summarise(total = sum(count))

# Mean with no range check and no median
avg_rent <- mean(df$rent)
print(paste("Average rent:", avg_rent))

# groupby without table() inspection
by_type <- df %>% group_by(housing_type) %>% summarise(n = n())

# Unexplained filter threshold
high_risk <- df %>% filter(rate > 0.15)

# Sentinel value filter — -999 may be missing data code
income_avg <- df %>%
  filter(income != -999) %>%
  summarise(mean_income = mean(income))

# Left join with no row count check
merged <- left_join(df, df2, by = "county")

# Join on string key (county_name is not an ID)
merged2 <- inner_join(df, df2, by = "county_name")

# Outer join with no anti-join check
left_result <- left_join(df, df2, by = "id")

# Hardcoded alpha threshold
result <- t.test(df$group_a, df$group_b)
if (result$p.value < 0.05) {
  print("Significant")
}

# Percentage with no denominator printed
pct <- sum(df$evicted) / nrow(df) * 100
cat(sprintf("Eviction rate: %.1f%%\n", pct))

# Aggregation with no na.rm
total_evictions <- sum(df$evictions)
